src/v2/src/item/syscall.rs

```diff
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[repr(C, align(8))]
+pub struct Header {
+    pub item: super::Header,
+    pub num: usize,
+}
```

Comment:
I would have expected:
```suggestion
pub struct Header {    
    pub num: usize,
    pub args: [usize; 6],
}
```

Reply:
This depends on what you define as "header", I'd say that `args` and `ret` are "payload" and, in fact, the way that payload is handled depends on the `num`, which is contained in the header.
For example, the host only needs to read 3 arguments from argv on a read and it should not read return values at all.